packages/protocol/src/abstracts/BaseRouter.sol

```diff
@@ -255,23 +265,23 @@ abstract contract BaseRouter is SystemAccessControl, IRouter {
         if (requestor != address(this)) {
           revert BaseRouter__bundleInternal_flashloanInvalidRequestor();
         }
-        _addTokenToList(asset);
+        tokensToCheck = _addTokenToList(asset, tokensToCheck);
```

Comment:
hey, can you point me to a test doing actions in flash loan child ?
I have some concerns with respect to how tokensToCheck would work for assets received from flashloan

Reply:
Sure, here: https://github.com/Fujicracy/fuji-v2/blob/efed33417d3d0245673a57133ec73e7b05363696/packages/protocol/test/forking/polygon/SimpleRouter.t.sol#L84

Reply:
cool thanks